actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -89,19 +96,28 @@ class ActionCable.Connection
 
     open: ->
       ActionCable.log("WebSocket onopen event")
-      @disconnected = false
-      @subscriptions.reload()
 
-    close: ->
+      if @isProtocolSupported()
+        ActionCable.log("WebSocket using '#{@getProtocol()}' subprotocol.")
+        @disconnected = false
+        @subscriptions.reload()
+      else
+        ActionCable.log("WebSocket got '#{unsupported_protocol}' as the chosen protocol. Stopping monitor and disconnecting.")
+        @disconnect(ActionCable.Events.Disconnected.ProtocolNotSupported)
```

Comment:
WebSockets are an implementation detail of Action Cable so it smells wrong to dispatch WebSocket-based data here. The only useful data for the client is wether or not AC is going to reconnect automatically or stop.